### PR TITLE
Added searchAndCountAll method to Profile

### PR DESCRIPTION
**Pull request by joefitter**
* Search will search combined first and last names
* Include pil.licenceNumber
* Include role.type - cast to TEXT for comparison

### CODE REVIEW

File: schema/profile.js [hunk 1]
```diff
@@ -26,6 +28,67 @@ module.exports = db => {
...
+          searchFullName(search),
+          { '$pil.licenceNumber$': { [Op.iLike]: `%${search}%` } },
+          // cast ENUM field to TEXT for search
+          db.where(db.cast(db.col('type'), 'TEXT'), { [Op.eq]: search })
+        ]
+      }
+      : {};
+
+    return Promise.all([
+      Profile.count(),
```
Comment (lennym) [inline schema/profile.js]: This only works because we only have one establishment with data. It will need to filter to the establishment.

Comment (joefitter) [reply schema/profile.js]: Aha, I'm glad you brought that up...

That's in the API, as it uses req.establishment.id. the where clauses get merged

Comment (joefitter) [reply schema/profile.js]: now _that_ is a very good point. 🎩

File: schema/project.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const { Op } = require('sequelize');
+const { merge } = require('lodash');
 const { STRING, ENUM, DATE, UUID, UUIDV1 } = require('sequelize');
```
Comment (lennym) [inline schema/project.js]: You're `require`-ing `sequelize` twice here. You should add `Op` to the other require call.

File: schema/profile.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+const { merge, chain, omit } = require('lodash');
+const { Op } = require('sequelize');
 const { UUID, UUIDV1, STRING, DATEONLY, TEXT } = require('sequelize');
```
Comment (lennym) [inline schema/profile.js]: Double `require` ☝️

**Comment by lennym** [comment]
The `Profile` bit of this is going to need a little work to handle the fact that profiles are now associated to establishments via a permission and not an `establishmentId` in the model.

File: schema/profile.js (hunk 3 above)
Comment (joefitter) [reply schema/profile.js]: OOPS

**Pull request closed without merging**